### ISSUE TITLE
Imports within a file with no 'package' or 'import' declarations

### ISSUE THREAD

**Issue by vkazanov** [dakrone/emacs-java-imports#4]
In Java (see [the grammar](https://docs.oracle.com/javase/specs/jls/se7/html/jls-18.html)) both `package` and `import` declarations are optional. This means that within such a file `java-imports-go-to-imports-start` won't do anything reasonable: it'll go to the beginning of the file and move forward by a few lines.

In my little example this meant going into the middle of a class and leaving an import there. 

This can be easily fixed, of course, although the solution should probably be discussed a bit. The simplest thing that comes to mind - just go to `point-min` and drop the import there if there were no `import` or `package`.  

Thank you

**Comment by dakrone** [comment]
I agree with going to `point-min` if no imports exist, I will look at making this change (or if you want to do a PR and beat me to it). Thanks for bringing this up!

**Comment by vkazanov** [comment]
Ok, I'll do it. Although there's still one thing I do not understand in the function.

Why do you need to do `(forward-line 2)`?

**Comment by vkazanov** [comment]
I've checked a neighboring function - `java-imports-add-import` - and it seems very customized towards a specific import order. Could you also describe it..?

**Comment by dakrone** [comment]
Here is an example of a file with the implied order:

```java
/**
 * License (optional)
 */

package org.writequit.package;

import org.writequit.foo.bar.Baz;  (1)
import org.writequit.foo.baz.Eggplant;

import java.util.List; (2)
import java.util.Map;

public class Thing {
    ...
}
```

Where `(1)` is non-JRE classes and `(2)` are java language-specific imports. the `(forward-line 2)` is to advance for the java language imports.

I'm totally open to making this configurable, I just happened to write the initial implementation to work with the [large java project](https://github.com/elastic/elasticsearch) I do most of my Java development on :)

**Comment by vkazanov** [comment]
Now this explains things in the code a bit, thank you. Basically right now the code is just following the convention you have in your main project (BTW, been using Elasticsearch a few years ago - great stuff).

The problem is that I am really struggling to imagine a way to find a place for insertion in a both generic and a simple way. 

I think the choice is the following: 

1. take one of the conventions available - and use it in the code (this is what you did).
2. use a dumb but working way to insert the import (just find the last import, and add the code after it) - this is what I would do if I was the author of the package :-)
3. allow a customization of sorts, with a primitive default function and a proper way to do customization needed. This is what should probably be done. 

What do you think?

**Comment by dakrone** [comment]
> What do you think? 

I would go with number 3, and do something like:

```emacs-lisp
(defcustom java-imports-find-import-line 'java-imports-go-to-import-location
  "Goes to the location an import should be added, takes one arg, the import"
  :type 'function
  :group 'java-imports)
```

Then people can customize it with a custom function (or multiple ones can be provided in the package)

**Comment by dakrone** [comment]
Resolved by #5 but I'm terrible at `magit-gh-pulls` so the merge commit got rebased out.
